### DISCUSSION TITLE
Get info unique item collection in custom hook

### DISCUSSION THREAD

**Discussion by kaiquefeng** [directus/directus#12340] [Archive]
Hello, can I get a specific item within a collection in custom hook with the function (item.create)? I'm trying to base myself on the documentation, but it's not well specified if this is possible

https://docs.directus.io/extensions/hooks/

**Comment by rijkvanzanten** [comment]
You can use `readOne(key: PrimaryKey, query?: Query)` from the `ItemsService` to read a single item 👍🏻 

```js
const service = new ItemsService('my-collection', { schema, accountability });

const item = await service.readOne(15);
```

**Reply by rijkvanzanten** [reply]
@br41nslug Would be nice to auto-generate those docs using TS Doc to Markdown, and then render it in VuePress or something like that (hint hint wink wink 😉)

**Reply by dmonterde** [reply]
Hi, where can i found documentation for ItemsService and the other API services? I searched the docs, and google but found nothing.

**Reply by rijkvanzanten** [reply]
They're not currently formally documented yet

**Reply by Rifqi-Irfansyah** [reply]
what if i want get info item collection, the unique is filter role. The data will be a lot, not one data

**Comment by dwh1121** [comment]
I have an example that can be successfully executed, for those who need it:

```typescript
import { defineHook } from '@directus/extensions-sdk';

export default defineHook(({ filter, action }, { services }) => {
	const { ItemsService } = services;

	action('<collection>.items.update', async (input, { schema, database }) => {
		console.log('<collection> Item updated!');
		console.log(input);

		const service = new ItemsService('<collection>', { schema, knex: database });
		const item = await service.readOne(input.keys[0]);
		console.log(item);
	});
});
```

**Comment by sharkfin009** [comment]
is it a crazy idea to install prisma in my extension, as an alternative until the docs are avail?
I can see the link/pivot table that i need to query for a relational query. Been reading the source code but time pressure is on me ;)

**Reply by br41nslug** [reply]
Not as beautiful as the official docs but i did create typedoc generated reference for the available services. https://github.com/br41nslug/directus-extension-reference

**Reply by sharkfin009** [reply]
thanks a mil @br41nslug
